### PR TITLE
Add firmware for CozyKeys Bloomer keyboard

### PR DESCRIPTION
**Pull request by pcewing**
## Description

Add firmware for the [CozyKeys Bloomer](https://github.com/cozykeys/bloomer) keyboard. There are two versions of the keyboard (v2 and v3) which are different enough that it makes sense to split them into their own directories, similar to the Speedo keyboard.

I will be submitting a separate pull request to the configurator repository later tonight to add the default keymaps there.

Tested by:
- Built and flashed the default keymap to both the v2 and v3 Bloomer
- Loaded the `info.json` for both v2 and v3 into the configurator and ensured the layout and ordering was correct

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by pcewing** [comment]
Hmm, from the CI error it looks like I need a `readme.md` in each version's directory as well?

**Comment by drashna** [comment]
Ideally, yeah.

**Review by drashna — APPROVED** [review]

**Comment by pcewing** [comment]
I added a `readme.md` to the two version sub-directories. I didn't want to copy/paste the instructions so instead I just linked to the `readme.md` in the parent directory which already contains instructions on how to build and flash the firmware.

File: keyboards/cozykeys/bloomer/readme.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+See the
+[build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and
+the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for
+more information. Brand new to QMK? Start with our
+[Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Flashing the Firmware
+
+See the QMK docs for handwiring a keyboard; there is a section with
+instructions on how to flash the *.hex* file to the Teensy 2.0 controller.
```
Comment (fauxpark) [inline keyboards/cozykeys/bloomer/readme.md:28]:
```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

## Flashing the Firmware

See the QMK docs for handwiring a keyboard; there is a section with instructions on how to flash the *.hex* file to the Teensy 2.0 controller.
```

File: keyboards/cozykeys/bloomer/readme.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Bloomer
+
+![Bloomer](http://assets.cozykeys.xyz/images/keyboards/bloomer/bloomer-angle-2_1600x1600.jpg)
```
Comment (fauxpark) [inline keyboards/cozykeys/bloomer/readme.md:3]:
```suggestion
![Bloomer](http://assets.cozykeys.xyz/images/keyboards/bloomer/bloomer-angle-2_800x800.jpg)
```

File: keyboards/cozykeys/bloomer/v2/readme.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# CozyKeys Bloomer v2
+
+![Bloomer](http://assets.cozykeys.xyz/images/keyboards/bloomer/bloomer-angle-2_1600x1600.jpg)
```
Comment (fauxpark) [inline keyboards/cozykeys/bloomer/v2/readme.md:3]:
```suggestion
![Bloomer](http://assets.cozykeys.xyz/images/keyboards/bloomer/bloomer-angle-2_800x800.jpg)
```

File: keyboards/cozykeys/bloomer/v3/readme.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# CozyKeys Bloomer v3
+
+![Bloomer](http://assets.cozykeys.xyz/images/keyboards/bloomer-v3/bloomer-v3-angle_1600x1600.jpg)
```
Comment (fauxpark) [inline keyboards/cozykeys/bloomer/v3/readme.md:3]:
```suggestion
![Bloomer](http://assets.cozykeys.xyz/images/keyboards/bloomer-v3/bloomer-v3-angle_800x800.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cozykeys/bloomer/readme.md (hunk 1 above)
Comment (pcewing) [reply keyboards/cozykeys/bloomer/readme.md:28]: These render to identical HTML and I like keeping line length short as is plays more nicely with my editor (vim). Is this necessary?

File: keyboards/cozykeys/bloomer/readme.md (hunk 2 above)
Comment (pcewing) [reply keyboards/cozykeys/bloomer/readme.md:3]: Is there a reason not to link to the higher res image?

Comment (fauxpark) [reply keyboards/cozykeys/bloomer/readme.md:3]: You can link to it but the embed itself should not be unreasonably large.

File: keyboards/cozykeys/bloomer/readme.md (hunk 1 above)
Comment (fauxpark) [reply keyboards/cozykeys/bloomer/readme.md:28]: In general, we don't hard wrap lines - screens are large enough these days that it shouldn't be needed.

File: keyboards/cozykeys/bloomer/v2/readme.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# CozyKeys Bloomer v2
+
+![Bloomer](http://assets.cozykeys.xyz/images/keyboards/bloomer/bloomer-angle-2_800x800.jpg)
+
+For more information on the Bloomer and how to build/flash the firmware, see
+the [readme.md](../readme.md) in the parent directory.
```
Comment (fauxpark) [inline keyboards/cozykeys/bloomer/v2/readme.md:6]:
```suggestion
For more information on the Bloomer and how to build/flash the firmware, see the [readme.md](../readme.md) in the parent directory.
```

File: keyboards/cozykeys/bloomer/v3/readme.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+# CozyKeys Bloomer v3
+
+![Bloomer](http://assets.cozykeys.xyz/images/keyboards/bloomer-v3/bloomer-v3-angle_800x800.jpg)
+
+For more information on the Bloomer and how to build/flash the firmware, see
+the [readme.md](../readme.md) in the parent directory.
```
Comment (fauxpark) [inline keyboards/cozykeys/bloomer/v3/readme.md:6]:
```suggestion
For more information on the Bloomer and how to build/flash the firmware, see the [readme.md](../readme.md) in the parent directory.
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cozykeys/bloomer/v2/rules.mk [hunk 7]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```
[the remark this answers is not in the document]
Comment (Erovia) [reply keyboards/cozykeys/bloomer/v2/rules.mk:10]: Similarly to the `config.h`, the `rules.mk` files could also be consolidated as `keyboards/cozykeys/bloomer/rules.mk`.
In this case, the `v2` and `v3` will need to have an empty `rules.mk` and `keyboards/cozykeys/bloomer/rules.mk` needs to have `DEFAULT_FOLDER = cozykeys/bloomer/v2`.

As this creates a new build target (`cozykeys/bloomer` that the mentioned option will make equal to `cozykeys/bloomer/v2`) I leave it up to you to decide to make the change or leave it as is.

File: keyboards/cozykeys/bloomer/v2/config.h [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+#endif
+
+// Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed
+#define DEBOUNCE 5
+
+// Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap
+#define LOCKING_SUPPORT_ENABLE
+// Locking resynchronize hack
+#define LOCKING_RESYNC_ENABLE
+
```
[the remark this answers is not in the document]
Comment (pcewing) [reply keyboards/cozykeys/bloomer/v2/config.h:59]: Ah good to know! I would actually prefer to keep the `MATRIX_ROW_PINS, MATRIX_COL_PINS, UNUSED_PINS` separate. I'm currently working on the v4 which will most likely have different pins as it no longer uses the ItsyBitsy MCU. Is it alright if I only consolidate the rest?

File: keyboards/cozykeys/bloomer/v2/rules.mk (hunk 7 above)
[the remark this answers is not in the document]
Comment (pcewing) [reply keyboards/cozykeys/bloomer/v2/rules.mk]: Will it be considered a breaking change in the future if I need to change the default? If so, I'd prefer not to do this just yet as I'd like to make v4 the default once it's ready and don't want the introduction of v4 firmware to have to go through the breaking change process.

File: keyboards/cozykeys/bloomer/v2/config.h (hunk 8 above)
[the remark this answers is not in the document]
Comment (Erovia) [reply keyboards/cozykeys/bloomer/v2/config.h:59]: Sure.

File: keyboards/cozykeys/bloomer/bloomer.h [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#ifdef KEYBOARD_cozykeys_bloomer_v2
+    #include "v2.h"
+#elif KEYBOARD_cozykeys_bloomer_v3
```
Comment (fauxpark) [inline keyboards/cozykeys/bloomer/bloomer.h:21]:
```suggestion
#if defined(KEYBOARD_cozykeys_bloomer_v2)
    #include "v2.h"
#elif defined(KEYBOARD_cozykeys_bloomer_v3)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by Erovia — APPROVED** [review]

**Pull request merged**